src/tracer.js

```diff
@@ -68,6 +68,23 @@ export default class Tracer {
         let binaryCodec = new BinaryCodec();
         this.registerInjector(opentracing.FORMAT_BINARY, binaryCodec);
         this.registerExtractor(opentracing.FORMAT_BINARY, binaryCodec);
+
+        this._setProcess();
+    }
+
+    _setProcess(): void {
+        let tags = [];
+        for (let key in this._tags) {
+            let value = this._tags[key];
+            if (this._tags.hasOwnProperty(key)) {
+                tags.push({ 'key': key, 'value': value });
+            }
+        }
+
+        this._reporter.setProcess({
+            'serviceName': this._serviceName,
+            'tags': Span._getThriftTags(tags)
+        });
```

Comment:
similar to comment in span.js, the thrift logic does not belong in Tracer. The API to the reporter can be `setProcess(serviceName, tags)`, and thrift construction can be deferred to as late as needed, e.g. in the sender, or at least the reporter (since interface to sender is already thrift)